### ISSUE TITLE
Named variables in .moor files?

### ISSUE THREAD

**Issue by PiN73** [simolus3/moor#922]
At the moment [colon-named variables (:id)](https://moor.simonbinder.eu/docs/using-sql/moor_files/#variables) generates method with positioned arguments:

```sql
myQuery:
    SELECT *
    FROM MyTable
    WHERE x_column = :x_value AND y_column = :y_value;
```

```dart
Selectable<MyQueryData> myQuery(String x_value, String y_value)
```

If the order of same-type variables in sql query change in the future, api will break.

Could there be required named variables?
```dart
Selectable<MyQueryData> myQuery({@required String x_value, @required String y_value})
```

**Comment by PiN73** [comment]
Also the variable names for Dart (not sql-string) may be converted to lowerCamelCase like column names?
```dart
Selectable<MyQueryData> myQuery({@required String xValue, @required String yValue})
```
Not sure about in-sql variables code style

**Comment by PiN73** [comment]
Or maybe arguments order in Dart could be the same as in .moor, if declared. I mean at the moment
```sql
myQuery(:x_value AS TEXT, :y_value AS TEXT):
    SELECT *
    FROM MyTable
    WHERE y_column = :y_value AND x_column = :x_value;
```
produces
```dart
Selectable<DistrictData> myQuery(String y_value, String x_value)
```
but could
```dart
Selectable<DistrictData> myQuery(String x_value, String y_value)
```

**Comment by Mike278** [comment]
This was suggested in #918 and I believe it's currently implemented on the develop branch

**Comment by simolus3** [comment]
> I believe it's currently implemented on the develop branch

Exactly. It requires the `named_parameters` build option for backwards compatibility, but this will be supported in the next moor version.

> Or maybe arguments order in Dart could be the same as in .moor, if declared

We sort by the index of the parameter. This usually matches the declaration order in the actual statement, but it doesn't respect parameter hints from moor files. Changing this would be a subtle breaking change, so I think the current behavior will have to do. It shouldn't be much of an issue with named parameters.

> Also the variable names for Dart (not sql-string) may be converted to lowerCamelCase like column names?

Hm, good point. We could have another build option for this.
